src/hash.h

```diff
@@ -112,23 +112,38 @@ class CHashWriter {
     int GetVersion() const { return nVersion; }
 
     void write(const char *pch, size_t size) {
-        ctx.Write({(const uint8_t *)pch, size});
+        ctx.Write((const uint8_t *)pch, size);
     }
 
-    // invalidates the object
+    /** Compute the double-SHA256 hash of all data written to this object.
+     *
+     * Invalidates this object.
+     */
     uint256 GetHash() {
         uint256 result;
-        ctx.Finalize(result);
+        ctx.Finalize(result.begin());
```

Comment:
Why does this need to change? Is it just to reset it?

Reply:
Again, different signatures:

```
void CHash256::Finalize(Span<uint8_t> output);
vs.
void CSHA256::Finalize(uint8_t hash[32]);
```

Should probably be refactored at some point, but not yet.